src/codegen/codegen_c_host.cc

```diff
@@ -0,0 +1,251 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ * \file codegen_c_host.cc
+ */
+#include <tvm/packed_func_ext.h>
+#include <vector>
+#include <string>
+#include "codegen_c_host.h"
+#include "build_common.h"
+
+namespace tvm {
+namespace codegen {
+
+CodeGenCHost::CodeGenCHost() {
+  module_name = GetUniqueName("__tvm_module_ctx");
```

Comment:
This would make the generated code necessarily dependent on C++, but we want to keep the option of pure C as well. If there's a reason to use the latter symbol, I can change it.